### ISSUE TITLE
proposal field should ideally be a list of strings

### ISSUE THREAD

**Issue by felipesanches** [fonttools/fontspector#23]
Checks such as **opentype/stat/ital_axis** have `proposal = "https://github.com/fonttools/fontbakery/issues/2934 and https://github.com/fonttools/fontbakery/issues/3668 and https://github.com/fonttools/fontbakery/issues/3669"`

The `proposal` field should instead be a list of URL strings to be easily consumed by sphinx (or whatever tool we decide to use for generating documentation) and also displayed on the user interface (for all reporters).

**Comment by simoncozens** [comment]
You're right; I have been holding on to this design choice out of stubbornness (because the majority of checks only have one proposal) but it does mess things up for checks with more than one.

**Comment by simoncozens** [comment]
Actually there is a more subtle reason why we can't do this. Maybe I tried in the past and couldn't make it work.

The issue is that we are defining the checks as `const Check` at compile time. The size of a `Check` must be known, so we can't express "one or possibly more than one string reference"; `proposal: [&'a str]` is not allowed. `proposal: Vec<&'a str>` is theoretically possible - but a `Vec` is a heap allocation at runtime; you can't just magic up a `Vec` at compile time. Same goes for other abstractions like `Box`.

So a struct with a variable number of static strings in it cannot be stored as a compile-time constant, hence the current layout.

**Comment by simoncozens** [comment]
It's ugly but I suppose we could use the JSON `metadata` field for this.

**Comment by simoncozens** [comment]
Lucy fixed it!
